File: app/src/main/java/com/example/travelo/map/SliderAdapter.java
package com.example.travelo.map;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

import com.bumptech.glide.Glide;
import com.example.travelo.R;
import com.smarteist.autoimageslider.SliderViewAdapter;

import java.util.List;

public class SliderAdapter extends SliderViewAdapter<SliderAdapter.SliderAdapterVH> {

    private String markerId;
    private Context context;
    private List images;

    public SliderAdapter(Context context, String markerId ,List images) {
        this.context = context;
        this.images = images;
        this.markerId = markerId;
    }

    @Override
    public SliderAdapterVH onCreateViewHolder(ViewGroup parent) {
        View inflate = LayoutInflater.from(parent.getContext()).inflate(R.layout.image_slider_layout_item, null);
        return new SliderAdapterVH(inflate);
    }

    @Override
    public void onBindViewHolder(SliderAdapterVH viewHolder, final int position) {

        Glide.with(viewHolder.itemView)
                .load(images.get(position))
                .into(viewHolder.imageViewBackground);

        //ImageView dodaj = viewHolder.itemView.findViewById(R.id.iv_auto_image_slider);
        //dodaj.setOnClickListener(new View.OnClickListener() {
        //    @Override
        //    public void onClick(View v) {
        //        if (position == getCount()-1)
        //            DatabaseUtils.Companion.addImageToStorage(context, markerId);
        //    }
        //});

    }

    @Override
    public int getCount() {
        //slider view count could be dynamic size
        return images.size();
    }

    class SliderAdapterVH extends SliderViewAdapter.ViewHolder {

        View itemView;
        ImageView imageViewBackground;

        public SliderAdapterVH(View itemView) {
            super(itemView);
            imageViewBackground = itemView.findViewById(R.id.iv_auto_image_slider);
            this.itemView = itemView;
        }
    }
}

File: app/src/main/java/com/example/travelo/map/SliderAdapter.kt
package com.example.travelo.map

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import com.bumptech.glide.Glide
import com.example.travelo.R
import com.smarteist.autoimageslider.SliderViewAdapter

class SliderAdapter(private val images: List<*>) : SliderViewAdapter<SliderAdapter.SliderAdapterVH>() {


    override fun onCreateViewHolder(parent: ViewGroup): SliderAdapterVH? {
        val inflate = LayoutInflater.from(parent.context).inflate(R.layout.image_slider_layout_item, null)
        return SliderAdapterVH(inflate)
    }

    override fun onBindViewHolder(viewHolder: SliderAdapterVH, position: Int) {
        Glide.with(viewHolder.itemView)
            .load(images[position])
            .into(viewHolder.imageViewBackground)
    }

    override fun getCount(): Int {
        return images.size
    }

    class SliderAdapterVH(var itemView: View) : ViewHolder(itemView) {
        var imageViewBackground: ImageView = itemView.findViewById(R.id.iv_auto_image_slider)

    }
}